packages/create-react-app/index.js

```diff
@@ -173,10 +173,14 @@ function install(packageToInstall, verbose, callback) {
 function run(root, appName, version, verbose, originalDirectory, template) {
   var packageToInstall = getInstallPackage(version);
   var packageName = getPackageName(packageToInstall);
-
   install(packageToInstall, verbose, function(code, command, args) {
     if (code !== 0) {
-      console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
+      console.log();
+      console.error('Aborting installation.', chalk.cyan(command + ' ' + args.join(' ')), 'has failed.');
+      console.log();
+      console.log('Deleting', chalk.cyan(appName), 'from', chalk.cyan(originalDirectory));
+      fs.removeSync(path.join(originalDirectory, appName));
```

Comment:
I've amended the approach. Should be somewhat safer. Still doesn't account for the "user created a file in the folder during installation" use case.